packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -3,6 +3,13 @@ import cdk = require('@aws-cdk/core');
 import { ContainerDefinition } from './container-definition';
 import { CfnTaskDefinition } from './ecs.generated';
 
+/**
+ * Regex pattern to check if it is an ECR image URL.
+ *
+ * @experimental
+ */
+const ECR_IMAGE_REGEX = /(^[a-zA-Z0-9][a-zA-Z0-9-_]*).dkr.ecr.([a-zA-Z0-9][a-zA-Z0-9-_]*).amazonaws.com(.cn)?\/.*/;
```

Comment:
Will this match images from other partitions? Eg china

Reply:
Yes, it can work for China. And this is the regex we use in ECS to determine whether a URI is ECR.